### PR TITLE
Add cache and schedule for ghaction

### PR DESCRIPTION
**Pull request by liangyuanpeng**
- Add github gradle cache for CI
- Add schedule to run ci every day

### CODE REVIEW

File: .github/workflows/build-pr.yml [hunk 1]
```diff
@@ -25,6 +25,8 @@ on:
     branches:
       - main
   workflow_dispatch:
+  schedule:
+  - cron: "0 0 * * *"
```
Comment (lburgazzoli) [inline .github/workflows/build-pr.yml:29]: what is the purpose for that ?

Comment (liangyuanpeng) [reply .github/workflows/build-pr.yml:29]: Run CI with schedule  for discover flack tests early, does make sence?

Comment (lburgazzoli) [reply .github/workflows/build-pr.yml:29]: this workflow only runs on PRs so I don't think it makes much sense

Comment (liangyuanpeng) [reply .github/workflows/build-pr.yml]: Moved schedule to main workflow,and testing for run snapshot exclude schedule.

**Comment by liangyuanpeng** [comment]
@lburgazzoli  PTAL,Thanks.

**Comment by lburgazzoli** [comment]
@liangyuanpeng rebase required

**Comment by liangyuanpeng** [comment]
@lburgazzoli  Done,Thanks.

**Pull request merged**